pkg/proc/core/linux_core.go

```diff
@@ -183,9 +258,16 @@ func readNote(r io.ReadSeeker) (*Note, error) {
 	descReader := bytes.NewReader(desc)
 	switch note.Type {
 	case elf.NT_PRSTATUS:
-		note.Desc = &LinuxPrStatus{}
-		if err := binary.Read(descReader, binary.LittleEndian, note.Desc); err != nil {
-			return nil, fmt.Errorf("reading NT_PRSTATUS: %v", err)
+		if machineType == EM_X86_64 {
+			note.Desc = &LinuxPrStatusAMD64{}
+			if err := binary.Read(descReader, binary.LittleEndian, note.Desc); err != nil {
```

Comment:
This if is the same in both branches, move it out. Leave only the assignment to `note.Desc` inside the two branches.